crates/uv/tests/sync.rs

```diff
@@ -162,3 +162,46 @@ fn frozen() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn empty() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [tool.uv.workspace]
+        members = []
+        "#,
+    )?;
+
+    // Running `uv sync` should generate an empty lockfile.
+    uv_snapshot!(context.filters(), context.sync(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: `uv sync` is experimental and may change without warning
+    warning: No `requires-python` value found in the workspace. Defaulting to `>=3.12`.
```

Comment:
Maybe we should throw a requires-python in there to reduce the noise?

Reply:
Not sure if we "can" since it's a virtual workspace (not a project). I'll try.

Reply:
Typically it would have at least one member.

Reply:
Yeah as soon as we add a `project` field it becomes non-virtual and the package itself gets included in the lockfile (making it non-empty).